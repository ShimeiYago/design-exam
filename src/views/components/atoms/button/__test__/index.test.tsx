import { shallow, ShallowWrapper } from 'enzyme';
import { Button, Props } from '..';

let wrapper: ShallowWrapper<Props>;

const props: Props = {
  children: 'login',
  color: 'basic',
  onClick: jest.fn(),
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Button {...props} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('fullWidth case', () => {
    wrapper.setProps({ fullWidth: true });
    expect(wrapper).toMatchSnapshot();
  });
});
