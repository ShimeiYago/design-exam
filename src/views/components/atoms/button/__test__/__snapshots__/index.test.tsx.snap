// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<button
  className="button basic"
  onClick={[MockFunction]}
>
  login
</button>
`;

exports[`Shallow Snapshot Tests fullWidth case 1`] = `
<button
  className="button basic fullWidth"
  onClick={[MockFunction]}
>
  login
</button>
`;
